### PR TITLE
Make 5 attempts at cleaning up sub-processes before giving up

### PR DESCRIPTION
**Pull request by Mesbah-Alam**
Added a loop around the existing logic to clean up child processes to make 5 attempts if cleanup fails. 
Related :  https://github.com/eclipse-openj9/openj9/issues/11557

Signed-off-by: Mesbah-Alam <Mesbah_Alam@ca.ibm.com>

### CODE REVIEW

**Comment by Mesbah-Alam** [comment]
Tested : https://openj9-jenkins.osuosl.org/job/Grinder_iteration_1/64/console

```
16:19:02  STF 20:15:14.933 - Killing processes: LT2
16:19:02  STF 20:15:14.933 -   o Process clean up attempt 1 for LT2 pid 17498508
16:19:02  STF 20:15:14.933 -   o Process LT2 pid 17498508 stop()
16:19:12  STF 20:15:24.934 -   o Process LT2 pid 17498508 terminate()
16:19:24  STF 20:15:34.935 -   o Process clean up attempt 2 for LT2 pid 17498508
16:19:24  STF 20:15:34.935 -   o Process LT2 pid 17498508 stop()
16:19:26  STF 20:15:38.935 -   o Process LT2 pid 17498508 killed
16:19:26  STF 20:15:38.936 - EXECUTE stage completed
16:19:26  STF 20:15:39.218 - 
16:19:26  STF 20:15:39.218 - ====================   T E A R D O W N   ====================
16:19:26  STF 20:15:39.218 - Running teardown: perl /home/jenkins/workspace/Grinder_iteration_1/aqa-tests/TKG/../TKG/output_16542008891973/TestIBMJlmRemoteClassNoAuth_0/20220602-201450-TestIBMJlmRemoteClassNoAuth/tearDown.pl
16:19:26  STF 20:15:39.324 - TEARDOWN stage completed
16:19:26  STF 20:15:39.335 - 
16:19:26  STF 20:15:39.335 - =====================   R E S U L T S   =====================
16:19:26  STF 20:15:39.335 - Stage results:
16:19:26  STF 20:15:39.336 -   setUp:     pass
16:19:26  STF 20:15:39.336 -   execute:   pass
16:19:26  STF 20:15:39.336 -   teardown:  pass
16:19:26  STF 20:15:39.336 - 
16:19:26  STF 20:15:39.336 - Overall result: PASSED
16:19:26  
16:19:26  TestIBMJlmRemoteClassNoAuth_0_PASSED
```

File: stf.core/scripts/stf/Commands.pm [hunk 1]
```diff
@@ -691,38 +691,49 @@ sub kill_process {
...
 
-	if($process->poll()) {
-		info("  o Process $process->{uid} pid $ppid terminate()");
-		$process->terminate();
-	}
+	while($attempt < 6) {
+		info("  o Process clean up attempt $attempt for $process->{uid} pid $ppid");
+		unless ($process->poll()) {
+			info("  o Process $process->{uid} pid $ppid is not running");
+			return;
```
Comment (pshipton) [inline stf.core/scripts/stf/Commands.pm]: Having a return here, in particular inside the loop, seems wrong. Possibly there needs to be some time after calling stop() before checking if the process is still running. The process could be found stopped after looping around. The later code echos any remaining output and returns a 0 or 1 code.

Comment (pshipton) [reply stf.core/scripts/stf/Commands.pm]: The later code also does `$process->{killed} = $TRUE;` which may not be desirable if stop() was never called.

Comment (Mesbah-Alam) [reply stf.core/scripts/stf/Commands.pm]: > Having a return here, in particular inside the loop, seems wrong.

Thanks for spotting! Moved the while loop below that check (of whether the process is already dead). 

 >  Possibly there needs to be some time after calling stop() before checking if the process is still running

There already exists a 10 second wait [inside the stop() subroutine](https://github.com/adoptium/STF/blob/master/stf.core/scripts/stf/ProcessMgmt/UNIX.pm#L295) - which is why I didn't introduce any additional wait after invocation of stop().

**Review by pshipton — APPROVED** [review]

**Pull request merged**